src/public/secure/admin.ejs

```diff
@@ -497,6 +560,187 @@
 
         function loadEventListeners() {
             const changeDetailsModal = new bootstrap.Modal('#changeDetailsModel');
+
+
+            ((() => {
+                function getMappingHtml(extensions, headers, isZip) {
+                    let retHtml = ``;
+                    for (let i = 0; i < extensions.length; i++) {
+                        const extension = extensions[i];
+                        const header = headers[i];
```

Comment:
I think this will bomb out if the length of headers is ever < extensions.  If they are bound together, maybe consider a array of tuples?

Reply:
It won't. The validation at this point will ensure that the arrays are akways of equal length.